sunpy/timeseries/timeseriesbase.py

```diff
@@ -432,9 +451,13 @@ def concatenate(self, others, same_source=False, **kwargs):
         # Concatenate the metadata and data.
         kwargs["sort"] = kwargs.pop("sort", False)
         meta = self.meta.concatenate([series.meta for series in others])
+        # TODO: refactor this to avoid going via DataFrames
```

Comment:
Merging this with this hack in place isn't something I am particularly happy about. Not without some substantive justification anyway.

Reply:
Why not?

Reply:
Mainly because it destroys all the benefits of being backed by Astropy Time if you cycle out and back in again.